snuba/subscriptions/scheduler.py

```diff
@@ -318,3 +385,24 @@ def find(
         if any(metric for metric in metrics if metric[1] > 0):
             for metric in metrics:
                 self.__metrics.increment(metric[0], metric[1], tags=metric[2])
+
+    def find_with_tick(
+        self, tick: Tick
+    ) -> Iterator[ScheduledTask[Tuple[Subscription, Tick]]]:
+        interval = tick.timestamps
+
+        subscriptions = self.__get_subscriptions()
+
+        for timestamp in range(
+            math.ceil(interval.lower.timestamp()),
+            math.ceil(interval.upper.timestamp()),
+        ):
+            for subscription in subscriptions:
+                task = self.__builder.get_task_v2((subscription, tick), timestamp)
+                if task is not None:
+                    yield task
+
+        metrics = self.__builder.reset_metrics()
+        if any(metric for metric in metrics if metric[1] > 0):
+            for metric in metrics:
+                self.__metrics.increment(metric[0], metric[1], tags=metric[2])
```

Comment:
Looking at how the `find` method is executed, it seems to me you could just replace the old `find` method with this one and pass the Tick in all cases. Where the `find` method is called you have both the interval and the Tick so you should be fine.
That would remove all this annoying code duplication. Basically all find methods are fully duplicated.

Reply:
Updated to remove the code duplication and use the new methods everywhere. Previously the `Tick` in the old pipeline was different to the new one, I had to make them the same - there are quite a few changes to the existing subscriptions consumer to make this work.